pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -180,3 +184,85 @@ type HTTPSDConfig struct {
 	// +optional
 	TLSConfig *v1.SafeTLSConfig `json:"tlsConfig,omitempty"`
 }
+
+// HTTPSDConfig defines a prometheus HTTP service discovery configuration
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#consul_sd_config
+// +k8s:openapi-gen=true
+type ConsulSDConfig struct {
+	// A valid string consisting of a hostname or IP followed by an optional port number.
+	// +kubebuilder:validation:MinLength:=1
+	// +required
+	Server string `json:"server,omitempty"`
```

Comment:
```suggestion
	Server string `json:"server"`
```

Reply:
We already have the `HostPort` struct for AlertmanagerConfig.
Maybe we can have a new `HostWithOptionalPort` struct? The downside is that we drift from the original Prometheus configuration format but we'll get a better/cleaner validation.

Reply:
I like that @simonpasquier but maybe we should have a kind of shared type where we can reuse this because I do believe other places may leverage those things.

WDYT?

If you think it's a good idea, where should I create this shared type? I'm bit lost yet in terms of where I should create things on the project

Reply:
If we'll only use this new struct for the Kubernetes SD config, it's fine to put it in `pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go`. If/when it gets reused, we can discuss a new location.

Reply:
@simonpasquier During this implementation I was wondering which kind of validation you have in mind, because looking at the host port structure, they are both strings, without any especial validation, since the port could be any port number or port name (when using k8s services) I can't see which kind of validation we could apply.

I'm afraid of this structure didn't provide any extra value for us (Of course we will have a meaningful type but without robust type validation).

WDYT?

Reply:
The main advantage from my point of view is that you avoid obvious issues like "foo:" or ":8080" but I agree that it might not be worth the effort for now...
I've looked into how the Consul library parses the configured address and it's more complicated than that:
https://github.com/hashicorp/consul/blob/7689a5ef2df8a92f476aabdb6ebd6e3696f8c39b/api/api.go#L722-L755
Also CRD validation rules can't really support all the possible [cases](https://cs.opensource.google/go/go/+/refs/tags/go1.20.5:src/net/url/url.go;l=215-220) for host values :shrug: 

We can at least validate that when defined (e.g. not nil), the server string isn't empty.

Reply:
Ok!
Makes sense, I'm thinking if not makes sense we use only one structure named Host where we wave an option field named port and apply all the validations against that.

If you think makes sense, we can address it on another issue since my idea is to change this in every place we're using plain strings to handle host values.

WDYT? @simonpasquier

Reply:
I agree that it should/can be addressed later. For now let's keep it simple (e.g. the `server` field should be of length > 0). Since the API is v1alpha1, we're still allowed to "break" (a bit) the interface.